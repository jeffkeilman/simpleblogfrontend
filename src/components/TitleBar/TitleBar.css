.TitleBar {
  display: block;
  line-height: 0;
  background-color: #ececec;
}