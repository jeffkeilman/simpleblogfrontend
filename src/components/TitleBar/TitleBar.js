import './TitleBar.css'
import InfoPanel from '../InfoPanel/InfoPanel'
import Face from '../Face/Face'

function TitleBar () {
  return (
    <header className='TitleBar'>
      <InfoPanel
        name='Jeff Keilman'
        githuburl='https://github.com/jeffkeilman'
        linkedinurl='https://www.linkedin.com/in/jeffkeilman'
      />
      <Face />
    </header>
  )
}

export default TitleBar
