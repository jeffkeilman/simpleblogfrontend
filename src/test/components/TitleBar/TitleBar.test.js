import React from 'react'
import renderer from 'react-test-renderer'
import TitleBar from '../../../components/TitleBar/TitleBar'
import { render } from '@testing-library/react'

describe('TitleBar', () => {
  // snapshot test
  it('renders correctly', () => {
    const tb = renderer.create(<TitleBar />).toJSON()
    expect(tb).toMatchSnapshot()
  })

  it('renders InfoPanel', () => {
    const { container } = render(<TitleBar />)
    expect(container.querySelector('.InfoPanel')).toBeInTheDocument()
  })

  it('renders Face', () => {
    const { container } = render(<TitleBar />)
    expect(container.querySelector('.Face')).toBeInTheDocument()
  })
})
