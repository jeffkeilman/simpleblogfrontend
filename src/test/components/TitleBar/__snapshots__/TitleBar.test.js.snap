// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar renders correctly 1`] = `
<header
  className="TitleBar"
>
  <section
    className="InfoPanel"
  >
    <section
      className="Description"
    >
      <p>
        Hello. My name is 
        <span>
          Jeff Keilman
        </span>
        .
      </p>
      <p>
        I am a 
        <span>
          software engineer
        </span>
         with a passion for 
        <span>
          social justice
        </span>
        .
      </p>
      <p
        className="LinkParagraph"
      >
        Here are some of my 
        <span>
          other links
        </span>
        :
      </p>
      <a
        href="https://github.com/jeffkeilman"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="GitHub link"
          className="GitHubLink"
          src="GitHubMarkLogo.png"
        />
      </a>
      <a
        href="https://www.linkedin.com/in/jeffkeilman"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="LinkedIn link"
          className="LinkedInLink"
          src="LinkedInLogo.png"
        />
      </a>
      <p>
        And here is my 
        <span>
          blog
        </span>
        .
      </p>
    </section>
  </section>
  <section
    className="Face"
  >
    <img
      alt="Author's face"
      src="Face.jpg"
    />
  </section>
</header>
`;
